src/lib/es2015.promise.d.ts

```diff
@@ -177,14 +177,7 @@ interface PromiseConstructor {
      * @param reason The reason the promise was rejected.
      * @returns A new rejected Promise.
      */
-    reject(reason: any): Promise<never>;
-
-    /**
-     * Creates a new rejected promise for the provided reason.
-     * @param reason The reason the promise was rejected.
-     * @returns A new rejected Promise.
-     */
-    reject<T>(reason: any): Promise<T>;
+    reject<T = never>(reason?: any): Promise<T>;

```

Comment:
I would not recommend making `reason` optional. Yes, `undefined` is permitted but it is generally a bad idea to throw `undefined` (which is effectively what this is doing). Requiring the parameter helps catch a certain class of errors, namely an "I forgot to fill in a reason" typo. However, if you really and truly want to create a promise rejected with `undefined`, you can still be explicit about it (i.e. `Promise.reject(undefined)`).

Other than that, I'm perfectly fine with collapsing the two declarations into a single declaration with a default type argument.

Reply:
But I feel requiring the reason parameter is not the purpose of #23416. And it doesn't reflect the ES spec. @mhegazy Do you want to make a reason required?

Reply:
After a quick review it seems that we allow the reason to be elided in the `reject` callback passed as part of the `Promise` constructor.